scripts/build_baremetal.sh

```diff
@@ -0,0 +1,87 @@
+#!/bin/sh
+
+# all.sh
+#
+# This file is part of mbed TLS (https://tls.mbed.org)
+#
+# Copyright (c) 2014-2016, ARM Limited, All Rights Reserved
+#
+# Purpose
+#
+# To build for bare metal with given toolchain
+#
+# Warning: the test is destructive. It includes various build modes and
+# configurations, and can and will arbitrarily change the current CMake
+# configuration. After this script has been run, the CMake cache will be lost
+# and CMake will no longer be initialised.
```

Comment:
I do not think you are using CMAKE in this script. Also, I do not see any changes to CMAKE to suggest that IAR could build with that build system